wordpress-mobile/WordPress-iOS
# Issue/8359 signup email entry

Fixes #8359

Needs Review: @nheagy

**To test:**
- Launch the app. Logout if necessary.
- Select 'Create a WordPress site'. 
- Verify the Email Entry view is displayed.

**Notes:**
- Launching the email signup process from 'Create a WordPress site' is temporary until the new Login Prologue is implemented.
- The design has a Back button. However, since this is the first page, there is nothing to go back to. So a Cancel button is displayed instead.

**Behavior:**
- The email validity is checked on-the-fly. The Next button is enabled when a valid email address is detected.
- If somehow you can Next with an invalid email, the error "Please enter a valid email address." is shown. (reused from existing Signup process)
- When Next, the email availability is checked. If the email is taken, the message "Sorry, that email address is already being used!" is displayed. (reused from existing Signup process)
- If there is an internal error when checking email availability, the error "Unable to verify the email address. Please try again later." is shown. (new)

When a valid email address is entered, this temporary alert appears:
<img width="270" alt="confirmation" src="https://user-images.githubusercontent.com/1816888/34267862-c6447de4-e63b-11e7-93da-a44256fde950.png">

**Designed:**
<img width="250" alt="designed" src="https://user-images.githubusercontent.com/1816888/34267886-de3c99ae-e63b-11e7-92a1-16c900e4d235.png">

**Implemented:**
![implemented](https://user-images.githubusercontent.com/1816888/34267942-1b5a01d2-e63c-11e7-9721-cd994168092e.png)

![email_taken](https://user-images.githubusercontent.com/1816888/34270187-8cfae312-e644-11e7-97c0-ee94997cbe57.png)
